specifyweb/frontend/js_src/lib/components/QueryBuilder/Export.tsx

```diff
@@ -125,6 +135,148 @@ export function QueryExportButtons({
     );
   }
 
+  async function exportKmlSelected(): Promise<void> {
+    const name = formatExportFileName('kml');
+
+    const selectedResults = results?.current?.map((row) =>
+      row !== undefined && f.has(selectedRows, row[0])
+        ? row?.slice(1).map((cell) => cell?.toString() ?? '')
+        : undefined
+    );
+
+    if (selectedResults === undefined) return undefined;
+
+    const filteredResults = filterArray(selectedResults);
+
+    const columnsName = fields
+      .filter((field) => field.isDisplay)
+      .map((field) =>
+        generateMappingPathPreview(baseTableName, field.mappingPath)
+      );
+
+
+    let placemarkTarget: any = [];
+
+    filteredResults?.forEach((result: any) => {
+      let dataTarget: any = [];
+
+      // <ExtendedData>
+      let extendedDataTarget: any = [];
+      
+      fields.forEach((field, index) => {
```

Comment:
this could be implemented using `.map()` too

see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

that would match the code style in the rest of the codebase and reduce the need for explicitly specifying typescript types